### PR TITLE
CAF-3056

### PR DESCRIPTION
**Pull request by Anthony-McGreevy**

### CODE REVIEW

File: utils/data-processing-cloud-based-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/cloudbasedtasksubmitter/Main.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Main entry point for cloud based task submitter application, reading in arguments and performing application function
+ */
+public final class Main
+{
+
+    public static void main(String[] args) throws ConfigurationException, IOException, CodecException, TimeoutException
+    {
+        new Main().run();
+        System.exit(0);
```
Comment (kploch) [inline utils/data-processing-cloud-based-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/cloudbasedtasksubmitter/Main.java]: Isn't that default? i.e. System.exit(0)?

File: utils/data-processing-cloud-based-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/cloudbasedtasksubmitter/Main.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    public void run() throws CodecException, IOException, ConfigurationException, TimeoutException
+    {
+        final String ROOT_REFERENCE = getEnvironmentValue(CloudTaskSubmitterConstants.ROOT_REFERENCE);
```
Comment (kploch) [inline utils/data-processing-cloud-based-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/cloudbasedtasksubmitter/Main.java]: Naming convention for variables....

File: utils/data-processing-family-task-submitter/nbactions.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kploch) [inline utils/data-processing-family-task-submitter/nbactions.xml]: I don't think this file should be in here... It's just some Netbeans project file, right? Update gitignore.

File: utils/data-processing-family-task-submitter/dependency-reduced-pom.xml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kploch) [inline utils/data-processing-family-task-submitter/dependency-reduced-pom.xml]: This shouldn't be checked in. Please update gitignore file.

File: utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/CloudTaskSubmitterConstants.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
+/*
```
Comment (kploch) [inline utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/CloudTaskSubmitterConstants.java]: Rename the file to Family...

File: utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/Main.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    public static void main(String[] args) throws ConfigurationException, IOException, CodecException, TimeoutException
+    {
+        new Main().run();
+        System.exit(0);
+    }
+
+    public void run() throws CodecException, IOException, ConfigurationException, TimeoutException
+    {
+        final String ROOT_REFERENCE = getEnvironmentValue(CloudTaskSubmitterConstants.ROOT_REFERENCE);
```
Comment (kploch) [inline utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/Main.java]: Naming convention for variables....

File: utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/Main.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        final ElasticQuery esQuery = new ElasticQuery(ROOT_REFERENCE);
+        final JSONObject familyJson = esQuery.createFamilyJson(ROOT_REFERENCE);
+        final DocumentExtractor docBuilder = new DocumentExtractor(familyJson);
```
Comment (kploch) [inline utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/Main.java]: If I'm reading the code right then you're getting a map of documents with fields in ElasticQuery class, then convert it to json and convert the json to map again in DocumentExtractor....

File: utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/elasticsearch/ElasticQuery.java [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    private final TransportClient transportClient;
+    private final String ROOT_REFERENCE;
+
+    public ElasticQuery(final String ROOT_REFERENCE) throws ConfigurationException
+    {
+        this.transportClient = getTransportClient();
+        this.ROOT_REFERENCE = ROOT_REFERENCE;
+    }
+
+    public JSONObject createFamilyJson(final String filename)
```
Comment (kploch) [inline utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/elasticsearch/ElasticQuery.java]: Why do you not just convert the hits and it's fields to Map<String,Multimap<String, String>>? You have fields and references...

File: utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/elasticsearch/ElasticQuery.java [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    private final String ELASTIC_TYPE = getEnvironmentValue("CAF_ELASTIC_TYPE", "document");
+    private final String ELASTIC_HOST_NAME = getEnvironmentValue("CAF_ELASTIC_HOSTNAME", "vagrant-mesos");
+    private final int ELASTIC_PORT = Integer.parseInt(getEnvironmentValue("CAF_ELASTIC_TRANSPORT_PORT", "9300"));
+    private final TransportClient transportClient;
+    private final String ROOT_REFERENCE;
+
+    public ElasticQuery(final String ROOT_REFERENCE) throws ConfigurationException
+    {
+        this.transportClient = getTransportClient();
+        this.ROOT_REFERENCE = ROOT_REFERENCE;
```
Comment (kploch) [inline utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/elasticsearch/ElasticQuery.java]: Field naming convention! ALL_CAPS is for constants only...

**Review by kploch — CHANGES_REQUESTED** [review]

File: utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/Main.java [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+    public static void main(String[] args) throws ConfigurationException, IOException, CodecException, TimeoutException
+    {
+        new Main().run();
+        System.exit(0);
+    }
+
+    public void run() throws CodecException, IOException, ConfigurationException, TimeoutException
+    {
+        final String rootReference = getEnvironmentValue(FamilyTaskSubmitterConstants.ROOT_REFERENCE);
```
Comment (kploch) [inline utils/data-processing-family-task-submitter/src/main/java/com/github/cafdataprocessing/utilities/familytasksubmitter/Main.java:45]: It should be possible to supply a reference with arguments. Environment variable here is not a good idea. If, let's say a script wanted to call this app, it'd need to set env variable each time before calling the application. It's not a configuration option. It's an execution parameter

**Review by kploch — CHANGES_REQUESTED** [review]

File: utils/pom.xml [hunk 11]
```diff
@@ -56,5 +56,6 @@
         <module>data-processing-task-receiver-container</module>
         <module>data-processing-task-submitter</module>
         <module>data-processing-task-submitter-container</module>
+        <module>data-processing-cloud-based-task-submitter</module>
```
Comment (kploch) [inline utils/pom.xml]: This should be renamed....

File: utils/data-processing-family-task-submitter/ReadMe.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
+##Data-Processing-Cloud-Based-Task-Submitter
```
Comment (kploch) [inline utils/data-processing-family-task-submitter/ReadMe.md]: This should be renamed. All instances of "Cloud" should be renamed.

File: utils/data-processing-family-task-submitter/pom.xml [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifest>
+                            <addClasspath>true</addClasspath>
+                            <mainClass>com.github.cafdataprocessing.utilities.cloudbasedtasksubmitter.Main</mainClass>
```
Comment (kploch) [inline utils/data-processing-family-task-submitter/pom.xml]: Wrong....

**Review by kploch — CHANGES_REQUESTED** [review]

**Comment by Anthony-McGreevy** [comment]
@kploch Updated to reflect comments

**Pull request merged**